coding/python_testing.rst

```diff
@@ -42,44 +41,149 @@ The important things to note in this example are:
 * When testing that an exception is raised always use :meth:`~unittest.TestCase.assertRaises` as a context manager, as shown in line 10 of the above example.
 * If a test method completes, the test passes; if it throws an uncaught exception the test has failed.
 
-Using a more advanced test runner
-=================================
+Supporting Pytest
+=================
 
 .. note::
-   `pytest`_ is not currently installed as part of a LSST stack installation.
-   It can be installed using :command:`conda install pytest` or :command:`pip install pytest`.
+  `pytest`_ and its plugins are standard stack EUPS packages and do not have to be installed separately.
+
+All LSST products that are built using :command:`scons` will execute Python tests using `pytest`_ so all tests should be written using it.
+`pytest`_ provides a much richer execution and reporting environment for tests and can be used to run multiple test files together.
 
-All tests should be written such that they are runnable using `pytest`_, a policy adopted in :jira:`RFC-69`.
-`pytest`_ provides a much richer execution and reporting environment for tests and can be used to run multiple tests files together.
-`pytest`_ test discovery is much more flexible than that provided by :mod:`unittest`, but LSST test files should not take advantage of that flexibility as it can lead to inconsistency in test reports that depend on the specific test runner.
+The `pytest`_ scheme for discovering tests inside Python modules is much more flexible than that provided by :mod:`unittest`, but LSST test files should not take advantage of that flexibility as it can lead to inconsistency in test reports that depend on the specific test runner, and it is required that an individual test file can be executed by running it directly with :command:`python`.
 In particular, care must be taken not to have free functions that use a ``test`` prefix or non-\ :class:`~unittest.TestCase` test classes that are named with a ``Test`` prefix in the test files.
 
+The tests/SConscript file
+-------------------------
+
+The behavior of `pytest`_ when invoked by :command:`scons` is controlled by the :file:`tests/SConscript` file.
+At minimum this file should contain the following to enable testing with automated test discovery:
+
+.. code-block:: python
+
+   from lsst.sconsUtils import scripts
+   scripts.BasicSConscript.tests(pyList=[])
+
+``pyList`` is used to specify which Python test files to run.
+Here the empty list is interpreted as "allow :command:`pytest` to automatically discover tests"
+:command:`pytest` will scan the directory tree itself to find tests and will run them all in together using the number of subprocesses matching the ``-j`` argument given to :command:`scons`.
```

Comment:
"all in together" is an unusual usage to this American.  Perhaps just "all together"?

Reply:
I agree. I was in the middle of saying "all in one process" and then realized that `xdist` means that is clearly not true.